From b2c90e5d5aff11c6f97f46b69f5005958a529347 Mon Sep 17 00:00:00 2001
[PATCH] Make openssl-1.0.1e build and pass tests

Summary:
Clean up the build files so that we can
build openssl-1.0.1e with buck.

---
 Android.mk       |  9 -----
 BUCK             |  5 ++-
 Crypto-config.mk | 29 +++++----------
 Crypto.mk        | 56 ++---------------------------
 5 files changed, 19 insertions(+), 83 deletions(-)

diff --git a/Android.mk b/Android.mk
index 46af985..94dd63c 100644
--- a/Android.mk
+++ b/Android.mk
@@ -1,17 +1,8 @@
 LOCAL_PATH := $(call my-dir)
 
-# Enable to be able to use ALOG* with #include "cutils/log.h"
-#log_c_includes += system/core/include
-#log_shared_libraries := liblog
-
-# These makefiles are here instead of being Android.mk files in the
-# respective crypto, ssl, and apps directories so
-# that import_openssl.sh import won't remove them.
 ifeq ($(strip $(TARGET_IS_64_BIT)),true)
   include $(LOCAL_PATH)/build-config-64.mk
 else
   include $(LOCAL_PATH)/build-config-32.mk
 endif
 include $(LOCAL_PATH)/Crypto.mk
-include $(LOCAL_PATH)/Ssl.mk
-include $(LOCAL_PATH)/Apps.mk
 
diff --git a/Crypto-config.mk b/Crypto-config.mk
index 5d01ce8..e27ab9a 100644
--- a/Crypto-config.mk
+++ b/Crypto-config.mk
@@ -545,14 +545,14 @@ common_src_files := \
   crypto/x509v3/v3_utl.c \
   crypto/x509v3/v3err.c \
 
-common_c_includes := \
-  . \
-  crypto \
-  crypto/asn1 \
-  crypto/evp \
-  crypto/modes \
-  include \
-  include/openssl \
+local_c_includes += \
+  $(LOCAL_PATH) \
+  $(LOCAL_PATH)/crypto \
+  $(LOCAL_PATH)/crypto/asn1 \
+  $(LOCAL_PATH)/crypto/evp \
+  $(LOCAL_PATH)/crypto/modes \
+  $(LOCAL_PATH)/include \
+  $(LOCAL_PATH)/include/openssl \
 
 arm_c_flags := \
   -DAES_ASM \
@@ -689,20 +689,9 @@ target_arch := unknown_arch
 endif
 
 target_c_flags    := $(common_c_flags) $($(target_arch)_c_flags) $(local_c_flags)
-target_c_includes := $(addprefix external/openssl/,$(common_c_includes)) $(local_c_includes)
+target_c_includes := $(local_c_includes)
 target_src_files  := $(common_src_files) $($(target_arch)_src_files)
 target_src_files  := $(filter-out $($(target_arch)_exclude_files), $(target_src_files))
 
-ifeq ($(HOST_OS)-$(HOST_ARCH),linux-x86)
-host_arch := x86
-else
-host_arch := unknown_arch
-endif
-
-host_c_flags    := $(common_c_flags) $($(host_arch)_c_flags) $(local_c_flags)
-host_c_includes := $(addprefix external/openssl/,$(common_c_includes)) $(local_c_includes)
-host_src_files  := $(common_src_files) $($(host_arch)_src_files)
-host_src_files  := $(filter-out $($(host_arch)_exclude_files), $(host_src_files))
-
 local_additional_dependencies += $(LOCAL_PATH)/Crypto-config.mk
 
diff --git a/Crypto.mk b/Crypto.mk
index ebb6e29..9106e9b 100644
--- a/Crypto.mk
+++ b/Crypto.mk
@@ -6,30 +6,6 @@ local_additional_dependencies := $(LOCAL_PATH)/android-config.mk $(LOCAL_PATH)/C
 
 include $(LOCAL_PATH)/Crypto-config.mk
 
-#######################################
-# target static library
-include $(CLEAR_VARS)
-include $(LOCAL_PATH)/android-config.mk
-
-LOCAL_SHARED_LIBRARIES := $(log_shared_libraries)
-
-# The static library should be used in only unbundled apps
-# and we don't have clang in unbundled build yet.
-ifeq ($(TARGET_ARCH),aarch64)
-$(info TODOAArch64: $(LOCAL_PATH)/Android.mk : Use NDK once it is available for AArch64)
-else
-LOCAL_SDK_VERSION := 9
-endif
-
-LOCAL_SRC_FILES += $(target_src_files)
-LOCAL_CFLAGS += $(target_c_flags)
-LOCAL_C_INCLUDES += $(target_c_includes)
-LOCAL_MODULE_TAGS := optional
-LOCAL_MODULE:= libcrypto_static
-LOCAL_ADDITIONAL_DEPENDENCIES := $(local_additional_dependencies)
-include $(BUILD_STATIC_LIBRARY)
-
-#######################################
 # target shared library
 include $(CLEAR_VARS)
 include $(LOCAL_PATH)/android-config.mk
@@ -45,40 +21,14 @@ else
 LOCAL_SDK_VERSION := 9
 endif
 LOCAL_LDFLAGS += -ldl
+LOCAL_LDLIBS += -lz
 
 LOCAL_SRC_FILES += $(target_src_files)
 LOCAL_CFLAGS += $(target_c_flags)
 LOCAL_C_INCLUDES += $(target_c_includes)
+LOCAL_EXPORT_C_INCLUDES := $(target_c_includes)
 LOCAL_MODULE_TAGS := optional
-LOCAL_MODULE:= libcrypto
+LOCAL_MODULE := libcryptox
 LOCAL_ADDITIONAL_DEPENDENCIES := $(local_additional_dependencies)
 include $(BUILD_SHARED_LIBRARY)
 
-#######################################
-# host shared library
-include $(CLEAR_VARS)
-include $(LOCAL_PATH)/android-config.mk
-LOCAL_SHARED_LIBRARIES := $(log_shared_libraries)
-LOCAL_SRC_FILES += $(host_src_files)
-LOCAL_CFLAGS += $(host_c_flags) -DPURIFY
-LOCAL_C_INCLUDES += $(host_c_includes)
-LOCAL_LDLIBS += -ldl
-LOCAL_MODULE_TAGS := optional
-LOCAL_MODULE:= libcrypto-host
-LOCAL_ADDITIONAL_DEPENDENCIES := $(local_additional_dependencies)
-include $(BUILD_HOST_SHARED_LIBRARY)
-
-########################################
-# host static library, which is used by some SDK tools.
-
-include $(CLEAR_VARS)
-include $(LOCAL_PATH)/android-config.mk
-LOCAL_SHARED_LIBRARIES := $(log_shared_libraries)
-LOCAL_SRC_FILES += $(host_src_files)
-LOCAL_CFLAGS += $(host_c_flags) -DPURIFY
-LOCAL_C_INCLUDES += $(host_c_includes)
-LOCAL_LDLIBS += -ldl
-LOCAL_MODULE_TAGS := optional
-LOCAL_MODULE:= libcrypto_static
-LOCAL_ADDITIONAL_DEPENDENCIES := $(local_additional_dependencies)
-include $(BUILD_HOST_STATIC_LIBRARY)
-- 
1.7.12.4 (Apple Git-37)

